contracts/staking/validatorShare/ValidatorShareFactory.sol

```diff
@@ -1,14 +1,28 @@
 pragma solidity ^0.5.2;
 
 import {ValidatorShareProxy} from "./ValidatorShareProxy.sol";
-
+import {ValidatorShare} from "./ValidatorShare.sol";
 
 contract ValidatorShareFactory {
     /**
     - factory to create new validatorShare contracts
    */
-
     function create(uint256 validatorId, address loggerAddress, address registry) public returns (address) {
-        return address(new ValidatorShareProxy(registry, validatorId, loggerAddress, msg.sender));
+        ValidatorShareProxy proxy = new ValidatorShareProxy(registry);
+
+        proxy.transferOwnership(msg.sender);
+
+        address proxyAddr = address(proxy);
+        (bool success, bytes memory data) = proxyAddr.call.gas(gasleft())(
+            abi.encodeWithSelector(
+                ValidatorShare(proxyAddr).initialize.selector, 
+                validatorId, 
+                loggerAddress, 
+                msg.sender
+            )
+        );
+        require(success, string(data));
```

Comment:
Why need to use `initialize` instead of using constructor? Because of how the proxy works? In that case, `registry` will be on proxy instead of implementation, right?

Reply:
Initialize is required because constructor already was invoked when we deploy implementation. We use proxy's storage, but layout lives in the implementation now. Hence, proxy's ctor is empty, storage also empty.

Before proxy inherited storage layout too which enables constructor with default storage values.

Yes, registry is default implementation in the proxy, as it was before. Then i obtain actual implementation from the registry and return it for delegatecall.